### ISSUE TITLE
torch.unique(dim=0, return_inverse=True) is returning the inverse indexes after the sort inside torch.unique(), not the original indexes

### ISSUE THREAD

**Issue by dongheuw** [pytorch/pytorch#64593]
## 🐛 Bug

torch.unique(dim=0, return_inverse=True) is returning the inverse indexes after sorting inside torch.unique(), not the actual original indexes. Actual use case is on a 2d tensor with dim=0 and return_inverse=True, but below is a simplified bug example on a 1d tensor.

## To Reproduce

import torch

a = torch.tensor([2, 3, 1, 1])
print(torch.unique(a, dim=0, return_inverse=True))

Output: (tensor([1, 2, 3]), tensor([0, 0, 1, 2]))

## Expected behavior

Output: (tensor([1, 2, 3]), tensor([1, 2, 0, 0]))

## Environment

PyTorch version: 1.9.0a0+gite7cd59c
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.19.6
Libc version: glibc-2.27

Python version: 3.8.8 (default, Apr 13 2021, 19:58:26)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-1056-azure-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.1.74
GPU models and configuration: GPU 0: Tesla P100-PCIE-16GB
Nvidia driver version: 455.23.05
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.3
[pip3] torch==1.9.0a0+gite7cd59c
[pip3] torch-scatter==2.0.8
[pip3] torchsort==0.1.6
[conda] blas                      1.0                         mkl
[conda] magma-cuda111             2.5.2                         1    pytorch
[conda] mkl                       2021.3.0           h06a4308_520
[conda] mkl-include               2021.3.0           h06a4308_520
[conda] mkl-service               2.4.0            py38h7f8727e_0
[conda] mkl_fft                   1.3.0            py38h42c9631_2
[conda] mkl_random                1.2.2            py38h51133e4_0
[conda] numpy                     1.20.3           py38hf144106_0
[conda] numpy-base                1.20.3           py38h74d4b33_0
[conda] torch                     1.9.0a0+gite7cd59c          pypi_0    pypi
[conda] torch-scatter             2.0.8                    pypi_0    pypi
[conda] torchsort                 0.1.6                    pypi_0    pypi
